src/rules/no-extraneous-dependencies.js

```diff
@@ -30,7 +29,64 @@ function extractDepFields(pkg) {
   };
 }
 
-function getDependencies(context, packageDir) {
+function getDependenciesMulti(packageDir, parentOptions, requireExact) {
+  const packageContent = {
+    dependencies: {},
+    devDependencies: {},
+    optionalDependencies: {},
+    peerDependencies: {},
+    bundledDependencies: [],
+  };
+
+  if (!depFieldCache.has(packageDir)) {
+    const packages = readPkgUp({
+      cwd: packageDir,
+      normalize: false,
+      multi: true,
+      requireExact,
+    });
+    if (packages.some(pkg => pkg.error)) {
+      throw packages.find(pkg => pkg.error).error;
+    }
+    const deps = packages.map(pkg => extractDepFields(pkg.pkg));
+    depFieldCache.set(packageDir, deps);
+  }
+  const packages = depFieldCache.get(packageDir);
+
+  if (parentOptions.consider) {
+    packages.forEach(pkg => Object.assign(packageContent.dependencies, pkg.dependencies));
+  } else {
+    Object.assign(packageContent.dependencies, packages[0].dependencies);
+  }
+
+  if (parentOptions.considerDev || (parentOptions.consider && parentOptions.considerDev !== false)) {
+    packages.forEach(pkg => Object.assign(packageContent.devDependencies, pkg.devDependencies));
+  } else {
+    Object.assign(packageContent.devDependencies, packages[0].devDependencies);
+  }
+
+  if (parentOptions.considerOptional || (parentOptions.consider && parentOptions.considerOptional !== false)) {
+    packages.forEach(pkg => Object.assign(packageContent.optionalDependencies, pkg.optionalDependencies));
```

Comment:
```suggestion
    packages.forEach(pkg => { Object.assign(packageContent.optionalDependencies, pkg.optionalDependencies); });
```
since we're not using the arrow functions' return value (same elsewhere)